accounts/models.py

```diff
@@ -164,9 +164,20 @@ def unread_notifications_count(self):
         return self.notifications.filter(read=False).count()
 
     def get_wards_display(self):
-        if not self.wards:
+        if not self.wards and not self.ward_principal:
             return "No wards"
+
         wards = cobrand.api.wards()
-        wards = {ward["gss"]: ward["name"] for ward in wards}
-        wards = [wards.get(w) for w in self.wards]
-        return ", ".join(wards)
+        ward_gss_to_name = {ward["gss"]: ward["name"] for ward in wards}
+
+        ward_principal_name = (
+            ward_gss_to_name.get(self.ward_principal) if self.ward_principal else None
+        )
+        assigned_ward_names = [ward_gss_to_name.get(w) for w in self.wards]
+
+        if ward_principal_name:
+            assigned_ward_names.remove(ward_principal_name)
```

Comment:
This assumes the ward principal is in the list (which it will be by the previous commit to the edit form, I see), but then line 167 has the new check that implies you could be in a situation where wards is not set but ward_principal is set? Don't mind either way, I could foresee a time where someone somehow edits the DB to give someone a principal but forget to add the assign, so do we want to check ward_principal_name is in assigned_ward_names here before removing it? Shrug :)